src/utils/update/psc.to.beneficial.owner.type.mapper.ts

```diff
@@ -0,0 +1,170 @@
+    first_name: psc.name_elements?.forename,
+    last_name: psc.name_elements?.surname,
+    nationality: psc.nationality,
+    // second_nationality: undefined,
+    date_of_birth: mapDateOfBirth(psc),
+    is_service_address_same_as_usual_residential_address: isSameAddress(service_address) ? yesNoResponse.Yes : yesNoResponse.No,
+    usual_residential_address: undefined,
+    service_address: service_address,
+    start_date: undefined,
+    is_on_sanctions_list: psc.is_sanctioned === true ? yesNoResponse.Yes : yesNoResponse.No,
+  };
+  mapNatureOfControl(psc, result, false);
+  return result;
+};
+
+export const mapPscToBeneficialOwnerOther = (psc: CompanyPersonWithSignificantControlResource): BeneficialOwnerOther => {
+  const service_address = mapAddress(psc.address);
+  const principal_address = mapAddress(psc.address);
+
+  const result: BeneficialOwnerOther = {
+    id: psc.links?.self,
+    name: psc.name,
+    principal_address: principal_address,
+    service_address: service_address,
+    is_service_address_same_as_principal_address: isSameAddress(service_address, principal_address) ? yesNoResponse.Yes : yesNoResponse.No,
+    legal_form: psc.identification?.legal_form,
+    law_governed: psc.identification?.legal_authority,
+    public_register_name: psc.identification?.place_registered,
+    registration_number: psc.identification?.registration_number,
+    is_on_register_in_country_formed_in: undefined,
+    start_date: undefined,
+    is_on_sanctions_list: psc.is_sanctioned === true ? yesNoResponse.Yes : yesNoResponse.No,
+  };
+  mapNatureOfControl(psc, result, false);
+  return result;
+};
+
+export const mapPscToBeneficialOwnerGov = (psc: CompanyPersonWithSignificantControlResource): BeneficialOwnerGov => {
+  const service_address = mapAddress(psc.address);
+  const principal_address = mapAddress(undefined);
+
+  const result: BeneficialOwnerGov = {
+    id: psc.links?.self,
+    name: psc.name,
+    principal_address: principal_address,
+    service_address: service_address,
+    is_service_address_same_as_principal_address: isSameAddress(service_address, principal_address) ? yesNoResponse.Yes : yesNoResponse.No,
+    legal_form: psc.identification?.legal_form,
+    law_governed: psc.identification?.legal_authority,
+    start_date: undefined,
+    is_on_sanctions_list: psc.is_sanctioned === true ? yesNoResponse.Yes : yesNoResponse.No,
+  };
+  mapNatureOfControl(psc, result, true);
+  return result;
+};
+
+const mapNatureOfControl = (psc: CompanyPersonWithSignificantControlResource, beneficialOwner: BeneficialOwnerIndividual| BeneficialOwnerOther | BeneficialOwnerGov, isBeneficialGov: boolean) => {
+  psc.natures_of_control?.forEach(natureType => {
+    const controlKind = natureTypeMap.get(natureType);
+    switch (controlKind) {
+        case 'BoNatureOfControl':
+          beneficialOwner.beneficial_owner_nature_of_control_types = natureOfControlTypeMap.get(natureType) as unknown as NatureOfControlType[];
```

Comment:
Don't need two as statements. I think this is just a string?

Reply:
Doesn't compile without first conversion to unknown using get(natureType). 
natureOfControlTypeMap[natureType] returns undefined

Reply:
It needs to be an array of NatureOfControlType so:

const natureOfControlType = natureOfControlTypeMap.get(natureType);
beneficialOwner.beneficial_owner_nature_of_control_types  = natureOfControlType ? [ natureOfControlType ] : [];

Reply:
same for other 2 below

Reply:
fixed